packages/sdk-manager/src/SdkManagerContext.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  EVMAssetTransfer,
+  Environment,
+  EvmFee,
+  Fungible,
+  Transfer
+} from '@buildwithsygma/sygma-sdk-core';
+import { consume, createContext, provide } from '@lit/context';
+import { LitElement, html } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import {
+  WalletManagerContext,
+  WalletManagerController
+} from '@builtwithsygma/sygmaprotocol-wallet-manager';
+import {
+  BaseProvider,
+  Web3Provider,
+  TransactionRequest
+} from '@ethersproject/providers';
+import { UnsignedTransaction } from '@ethersproject/transactions';
+
+export type SdkManagerStatus =
+  | 'idle'
+  | 'initialized'
+  | 'transferCreated'
+  | 'approvalsCompleted'
+  | 'deposited'
+  | 'completed';
+
+export type SdkManagerState = {
+  assetTransfer: EVMAssetTransfer;
+  status: SdkManagerStatus;
+  transfer?: Transfer<Fungible>;
+  fee?: EvmFee;
+  approvalTxs?: UnsignedTransaction[];
+  depositTx?: UnsignedTransaction;
+
+  initialize: (provider: BaseProvider, env?: Environment) => Promise<void>;
+  createTransfer: (
+    provider: BaseProvider,
+    destinationChainId: number,
+    destinationAddress: string,
+    resourceId: string,
+    amount: string
+  ) => Promise<void>;
+
+  performApprovals(provider: Web3Provider): Promise<void>;
+  performDeposit(provider: Web3Provider): Promise<void>;
+};
+
+export const SdkManagerContext = createContext<SdkManagerState | undefined>(
+  'sdk-context'
+);
+
+export class SdkManager implements SdkManagerState {
+  assetTransfer: EVMAssetTransfer;
+  status: SdkManagerStatus;
+  transfer?: Transfer<Fungible>;
+  fee?: EvmFee;
+  approvalTxs?: UnsignedTransaction[];
+  depositTx?: UnsignedTransaction;
+
+  constructor() {
+    this.assetTransfer = new EVMAssetTransfer();
+    this.status = 'idle';
+  }
+
+  async initialize(
+    provider: BaseProvider,
+    env: Environment = Environment.MAINNET
+  ) {
+    await this.assetTransfer.init(provider, env);
+    this.status = 'initialized';
+  }
+
+  async createTransfer(
+    provider: BaseProvider,
+    destinationChainId: number,
+    destinationAddress: string,
+    resourceId: string,
+    amount: string
+  ) {
+    const sourceAddress = await (provider as Web3Provider)
+      .getSigner()
+      .getAddress();
+
+    const transfer = await this.assetTransfer.createFungibleTransfer(
```

Comment:
if this method creates the transfer, it should only do that. Can we move the `getFee` call and the other to separate methods? Wdyt?

Reply:
oh wait, whats the diff between building the approvals here for instance and then on the `performApprovals` method below?

Reply:
maybe the same parts can be extracted into a method like `postTransferCreate` or something like that